mountpoint-s3-crt/src/s3/client.rs

```diff
@@ -533,6 +538,12 @@ impl MetaRequest {
             aws_s3_meta_request_cancel(self.inner.as_ptr());
         }
     }
+
+    /// Write a chunk of data and indicate whether it is the last. If invoked before the previous
+    /// write completed, or after setting `eof` to `true`, will return an AWS_ERROR_INVALID_STATE error.
+    pub fn write<'a>(&'a mut self, slice: &'a [u8], eof: bool) -> MetaRequestWrite<'a> {
```

Comment:
nit: I think this can just be
```suggestion
    pub fn write(&mut self, slice: &[u8], eof: bool) -> MetaRequestWrite<'_> {
```

Reply:
No, we also need to tie result to the lifetime of the slice.